### DISCUSSION TITLE
Possibly add a way to set permissions for slash commands

### DISCUSSION THREAD

**Discussion by Skittles3920** [discordjs/discord.js#7741] [Ideas]
I am not too sure if the Discord API allows it but would it be possible to add permissions to a slash command similarly to how you can add specific roles/users?

**Comment by monbrey** [comment]
This isn't supported by the API in the current implementation of permissions, but its a very popular request and has been considered by Discord.

**Comment by ff2400t** [comment]
sorry if am being an idiot as i just started making a bot two days ago but this can be used to for this purpose right
`interaction.memberPermissions.has(Permissions.FLAGS.MANAGE_EMOJIS_AND_STICKERS)`

**Reply by almostSouji** [reply]
Correct, this is the only way to limit commands based on permission bits at the moment. However, it moves the point of failure for the end-user. With roles and user limitations, the client itself makes the command unavailable to be used in the first place, and the API checks against it to prevent rogue requests from user-bots.

With this approach, you'd need to (potentially ephemerally) respond to inform the user that they cannot use the command. You could of course also just not respond to the user at all, however, that makes it seem like your bot is down, rather than a permission configuration being the cause.
